BulkyWeb/Areas/Admin/Controllers/CategoryController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Bulky.DataAccess.Data;
+using Bulky.DataAccess.Repository.IRepository;
+using Bulky.Models;
+using Microsoft.AspNetCore.Mvc;
+
+namespace BulkyWeb.Areas.Admin.Controllers
+{
+    [Area("Admin")]
+    public class CategoryController : Controller
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        public CategoryController(IUnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+        }
+        public IActionResult Index()
+        {
+            try
+            {
+                List<Category> objCategoryList = _unitOfWork.Category.GetAll().ToList();
+                return View(objCategoryList);
+            }
+            catch (Exception ex)
+            {
+                TempData["error"] = "Category NotFound";
```

Comment:
Same here the tempdata error message should not be generic. It should resemble what the actual error is. This index is bringing in all the categories so it should be something like No categories found